kyo-core/shared/src/test/scala/kyo/AsyncTest.scala

```diff
@@ -529,4 +557,176 @@ class AsyncTest extends Test:
+                yield assert(result.isPanic)
+                end for
+            }
+        }
+
+        "use" - {
+            "success" in run {
+                val fiber = Fiber.success[Nothing, Int](42)
+                for
+                    result <- fiber.use(x => x * 2)
+                yield assert(result == 84)
+            }
+
+            "failure" in run {
+                val ex    = new Exception("Test exception")
+                val fiber = Fiber.fail[Exception, Int](ex)
+                for
+                    result <- Abort.run(fiber.use(x => x * 2))
+                yield assert(result.failure.contains(ex))
+            }
+
+            "exception in use function" in run {
+                val fiber = Fiber.success[Nothing, Int](42)
+                for
+                    result <- Abort.run[Throwable](fiber.use(_ => throw new RuntimeException("Use exception")))
+                yield assert(result.isPanic)
+            }
+        }
+
+        "useResult" - {
+            "success" in run {
+                val fiber = Fiber.success[Nothing, Int](42)
+                for
+                    result <- fiber.useResult(r => r.map(_ * 2))
+                yield assert(result == Result.success(84))
+            }
+
+            "failure" in run {
+                val ex    = new Exception("Test exception")
+                val fiber = Fiber.fail[Exception, Int](ex)
+                for
+                    result <- fiber.useResult(r => r.mapFail(_.getMessage))
+                yield assert(result == Result.fail("Test exception"))
+            }
+
+            "exception in useResult function" in run {
+                val fiber = Fiber.success[Nothing, Int](42)
+                for
+                    result <- Abort.run[Throwable](fiber.useResult(_ => throw new RuntimeException("UseResult exception")))
+                yield assert(result.isFail)
+            }
+        }
+
+        "onComplete" - {
+            "already completed" in run {
+                var completed = false
+                val fiber     = Fiber.success(42)
+                for
+                    _ <- fiber.onComplete(_ => IO { completed = true })
+                yield assert(completed)
+                end for
+            }
+
+            "pending" in run {
+                var completed = Maybe.empty[Result[Nothing, Int]]
+                for
+                    fiber <- Promise.init[Nothing, Int]
+                    _     <- fiber.onComplete(v => IO { completed = Maybe(v) })
+                    notCompletedYet = completed
+                    _ <- fiber.complete(Result.success(42))
+                    completedAfterWait = completed
+                yield
+                    assert(notCompletedYet.isEmpty)
+                    assert(completedAfterWait == Maybe(Result.success(42)))
+                end for
+            }
+        }
+
+        "block" - {
```

Comment:
Should we have Fiber specific tests for interrupt?

Reply:
Given that `Fiber` is just wrapper of `IOPromise`, I don't think we need to replicate the tests. There are a couple of tests in this file that use `Fiber.interrupt` as well.